pkg/provider/provider_test.go

```diff
@@ -0,0 +1,119 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	_ "github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestProvider_SetData(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		key := []byte{1, 1, 1}
+		value := []byte{0, 1, 3}
+		encodedValue := []byte{0, 1, 3}
```

Comment:
I think it would be more explicit to make `value` and `encodedValue` to have different values because in that case it could found issues with passing `value` instead of `encodedValue` for method `Decode`.